lib/auth.js

```diff
@@ -0,0 +1,70 @@
+module.exports = loopbackAuth;
+
+function loopbackAuth(prefix) {
+  var props = ['accessTokenId', 'currentUserId'],
+      _sessionStorage,
+      _localStorage;
+
+  prefix = (prefix || 'loopback') + ':';
+
+  var auth = {
+    save: save,
+    setUser: setUser,
+    clearUser: clearUser,
+    clearStorage: clearStorage
+  };
+
+  if (typeof window !== 'undefined') {
+    _localStorage = window.localStorage;
```

Comment:
I don't think storage belongs in this module. All we need to do in the connector itself is ensure the token and any other metadata required by authentication is set on the context and or request. Something like this:

Everything else is really outside the domain of this module. I think it actually belongs in the LoopBack `AccessToken` or `User` models.

I think we just need something like this in the connector itself:

```js
// the dataSource
var remote = loopback.createDataSource();

// set the accessToken to be sent along with each request / invocation
remote.setAccessToken(accessToken);

// set the accessToken during creation
var remote = loopback.createDataSource({
  connector: 'remote',
  url: 'http://localhost:3000/api',
  accessToken: accessToken
});
```